src/audio/eq_iir.c

```diff
@@ -53,82 +53,69 @@
-#define tracev_eq_iir(__e) tracev_event(TRACE_CLASS_EQ_IIR, __e)
-#define trace_eq_iir_error(__e) trace_error(TRACE_CLASS_EQ_IIR, __e)
+#define trace_eq(__e) trace_event(TRACE_CLASS_EQ_IIR, __e)
+#define tracev_eq(__e) tracev_event(TRACE_CLASS_EQ_IIR, __e)
+#define trace_eq_error(__e) trace_error(TRACE_CLASS_EQ_IIR, __e)
 
-/* src component private data */
+/* IIR component private data */
 struct comp_data {
 	struct sof_eq_iir_config *config;
 	uint32_t period_bytes;
 	struct iir_state_df2t iir[PLATFORM_MAX_CHANNELS];
 	void (*eq_iir_func)(struct comp_dev *dev,
-		struct comp_buffer *source,
-		struct comp_buffer *sink,
-		uint32_t frames);
+			    struct comp_buffer *source,
+			    struct comp_buffer *sink,
+			    uint32_t frames);
 };
 
 /*
  * EQ IIR algorithm code
  */
 
+static void eq_iir_passthrough(struct comp_dev *dev,
+			       struct comp_buffer *source,
+			       struct comp_buffer *sink,
+			       uint32_t frames)
+{
+	int32_t *src = (int32_t *)source->r_ptr;
+	int32_t *dest = (int32_t *)sink->w_ptr;
+	int nch = dev->params.channels;
+	int n = frames * nch;
+
+	memcpy(dest, src, n * sizeof(int32_t));
+}
+
 static void eq_iir_s32_default(struct comp_dev *dev,
-	struct comp_buffer *source, struct comp_buffer *sink, uint32_t frames)
+			       struct comp_buffer *source,
+			       struct comp_buffer *sink,
+			       uint32_t frames)
+
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
+	struct iir_state_df2t *filter;
+	int32_t *src = (int32_t *)source->r_ptr;
+	int32_t *snk = (int32_t *)sink->w_ptr;
+	int32_t *x;
+	int32_t *y;
 	int ch;
-	int n;
-	int n_wrap_src;
-	int n_wrap_snk;
-	int n_wrap_min;
-	int32_t *src = (int32_t *) source->r_ptr;
-	int32_t *snk = (int32_t *) sink->w_ptr;
+	int i;
 	int nch = dev->params.channels;
-	int32_t *x = src + nch - 1;
-	int32_t *y = snk + nch - 1;
 
 	for (ch = 0; ch < nch; ch++) {
-		n = frames * nch;
+		filter = &cd->iir[ch];
 		x = src++;
 		y = snk++;
-		while (n > 0) {
-			n_wrap_src = (int32_t *) source->end_addr - x;
-			n_wrap_snk = (int32_t *) sink->end_addr - y;
-			n_wrap_min = (n_wrap_src < n_wrap_snk) ?
-				n_wrap_src : n_wrap_snk;
-			if (n < n_wrap_min) {
-				/* No circular wrap need */
-				while (n > 0) {
-					*y = iir_df2t(&cd->iir[ch], *x);
-					x += nch;
-					y += nch;
-					n -= nch;
-				}
-			} else {
-				/* Wrap in n_wrap_min/nch samples */
-				while (n_wrap_min > 0) {
-					*y = iir_df2t(&cd->iir[ch], *x);
-					x += nch;
-					y += nch;
-					n_wrap_min -= nch;
-					n -= nch;
-				}
-				/* Check both source and destination for wrap */
-				if (x > (int32_t *) source->end_addr)
-					x = (int32_t *)
-					((size_t) x - source->size);
-				if (snk > (int32_t *) sink->end_addr)
-					y = (int32_t *)
-					((size_t) y - sink->size);
-			}
+		for (i = 0; i < frames; i++) {
+			*y = iir_df2t(filter, *x);
+			x += nch;
+			y += nch;
 		}
-
 	}
 }
 
 static void eq_iir_free_parameters(struct sof_eq_iir_config **config)
 {
-	if (*config != NULL)
+	if (config)
```

Comment:
you sure about this one? shouldn't it be if (*config)?

Reply:
Yes, thanks! You've got a hawk eye for bugs!